pymc_marketing/customer_choice/mnl_logit.py

```diff
@@ -0,0 +1,586 @@
+        A list of formulas specifying how to model the utility of
+        each product alternative. The formulas should be in Wilkinson
+        style notation and allow the target product to be specified as
+        as a function of the alternative specific attributes and the individual
+        specific attributes:
+        target_product ~ target_attribute1 + target_attribute2 | individual_attribute
+
+    depvar : str
+        The name of the dependent variable in the choice_df.
+
+    covariates : list of str
+        Covariate names (e.g., ['X1', 'X2']).
+
+    model_config : dict, optional
+        Model configuration. If None, the default config is used.
+
+    sampler_config : dict, optional
+        Sampler configuration. If None, the default config is used.
+
+    Notes
+    -----
+    Example:
+    -------
+    The format of `choice_df`:
+
+        +------------+------------+------------+------------+------------+
+        | Depvar     | alt_1_X1   | alt_1_X2   | alt_2_X1   | alt_2_X2   |
+        +============+============+============+============+============+
+        | alt_1      | 2.4        | 4.5        | 5.4        | 6.7        |
+        +------------+------------+------------+------------+------------+
+        | alt_2      | 3.5        | 6.7        | 2.3        | 8.9        |
+        +------------+------------+------------+------------+------------+
+
+    Example `utility_equations` list:
+
+    >>> utility_equations = [
+    ...     'alt_1 ~ X1_alt1 + X2_alt1 | income',
+    ...     'alt_2 ~ X1_alt2 + X2_alt2 | income',
+    ...     'alt_3 ~ X1_alt3 + X2_alt3 | income'
+    ... ]
+
+    """
+
+    _model_type = "Multinomial Logit Model"
+    version = "0.1.0"
+
+    def __init__(
+        self,
+        choice_df: pd.DataFrame,
+        utility_equations: list[str],
+        depvar: str,
+        covariates: list[str],
+        model_config: dict | None = None,
+        sampler_config: dict | None = None,
+    ):
+        self.choice_df = choice_df
+        self.utility_equations = utility_equations
+        self.depvar = depvar
+        self.covariates = covariates
+
+        model_config = model_config or {}
+        model_config = parse_model_config(model_config)
+
+        super().__init__(model_config=model_config, sampler_config=sampler_config)
+
+    @property
+    def default_model_config(self) -> dict:
+        """Default model configuration.
+
+        This is a Categorical likelihood, Normal intercept,
+        and a Normal vector of beta coefficients
+
+        Returns
+        -------
+        dict
+            The default model configuration.
+
+        """
+        alphas = Prior("Normal", mu=0, sigma=5, dims="alts")
+        betas = Prior("Normal", mu=0, sigma=1, dims="alt_covariates")
+        betas_fixed = Prior("Normal", mu=0, sigma=1, dims=("alts", "fixed_covariates"))
+
+        return {
+            "alphas_": alphas,
+            "betas": betas,
+            "betas_fixed_": betas_fixed,
```

Comment:
Why do some have suffix of _ and others do not?

Reply:
I need to 0 out one of the alternatives alphas_ --> alpha and I was doing the same for betas_fixed_ for parameter identification. 

e.g alpha = pm.Deterministic('alpha', pt.set_subtensor(alphas_, -1, 0))....